extensions/wikia/TemplateClassification/specials/templates/TemplatesSpecialController_index.php

```diff
@@ -0,0 +1,52 @@
+<p>
+	<form method="GET" action="">
+		<label for="template">
+			<?= wfMessage( 'template-classification-special-find' )->escaped() ?>
+			<input name="template" type="text" value="<?= Sanitizer::encodeAttribute( $templateName ) ?>" />
+		</label>
+		<?php if ( !empty( $groups ) ): ?>
+			<select name="type">
+				<?php foreach( $groups as $group ): ?>
+					<option value="<?= Sanitizer::encodeAttribute( $group ) ?>" <?php if( $group == $type ): ?>selected<?php endif ?>><?= wfMessage( 'template-classification-type-' . $group )->escaped() ?></option>
+				<?php endforeach ?>
+			</select>
+		<?php endif ?>
+		<input type="submit" value="<?= wfMessage( 'template-classification-special-search' )->escaped() ?>">
+	</form>
+</p><br/>
+<?php if ( !empty( $templates ) ): ?>
+<table class="templates-hq article-table">
+	<tr>
+		<th><?= wfMessage( 'template-classification-special-page-header' )->escaped() ?></th>
+		<th><?= wfMessage( 'template-classification-special-used-header' )->escaped() ?></th>
+		<th><?= wfMessage( 'template-classification-special-type-header' )->escaped() ?></th>
+	</tr>
+	<?php foreach( $templates as $template ): ?>
+		<?php $groupName = wfMessage( 'template-classification-type-' . $type )->escaped(); ?>
```

Comment:
The variable here is `$type` which isn't part of the loop, so can this be defined before the loop so `wfMessage` isn't called for each template or should the variable be something specific to the template?